packages/cli/src/runIOS/runIOS.js

```diff
@@ -308,28 +307,79 @@ function buildProject(
       }
     });
     buildProcess.stderr.on('data', data => {
-      console.error(data.toString());
+      errorOutput += data;
     });
     buildProcess.on('close', code => {
       if (xcpretty) {
         xcpretty.stdin.end();
       }
-      // FULL_PRODUCT_NAME is the actual file name of the app, which actually comes from the Product Name in the build config, which does not necessary match a scheme name,  example output line: export FULL_PRODUCT_NAME="Super App Dev.app"
-      const productNameMatch = /export FULL_PRODUCT_NAME="?(.+).app"?$/m.exec(
-        buildOutput
-      );
-      if (
-        productNameMatch &&
-        productNameMatch.length &&
-        productNameMatch.length > 1
-      ) {
-        return resolve(productNameMatch[1]); // 0 is the full match, 1 is the app name
+      if (code !== 0) {
+        reject(
+          new ProcessError(
+            [
+              `Failed to build iOS project.`,
+              `We ran "xcodebuild" command but it exited with error code ${code}.`,
+              `To debug build logs further, consider building your app with Xcode.app, by opening ${
+                xcodeProject.name
+              }`,
+            ].join(' '),
+            errorOutput
+          )
+        );
+        return;
```

Comment:
is this return necessary after we `reject`ed?

Reply:
It's not - just my personal preference of early returns over /else clause (visual appearance).

Reply:
I think I'll update it to if/else here 👌